sdl2promela/translator.py

```diff
@@ -544,66 +559,192 @@ def __get_exit_procedures(
     return list(reversed(result))
 
 
-def __generate_input_function(
-    context: Context, input: sdlmodel.Input
-) -> promelamodel.Inline:
-    builder = InlineBuilder()
-    builder.withName(__get_input_function_name(context, input))
-    blockBuilder = BlockBuilder(promelamodel.BlockType.BLOCK)
-    for parameter in input.parameters:
-        builder.withParameter(__get_parameter_name(parameter.target_variable))
-
-    switch_builder = SwitchBuilder()
-
-    state_variable_name = __get_state_variable_name(context)
+def __generate_execute_transition(
+    context: Context,
+    statename: str,
+    input: sdlmodel.Input,
+    builder: SwitchBuilder,
+    parentstate: Optional[str],
+) -> bool:
+    """
+    Generate conditional call to execute transition inline.
+    This function checks if the Input contains a transition_id for the statename
+    and generates a call to the execute transition inline.
+    Also, the calls to exit procedures are generated if requires.
+
+    :param context: Global translator context
+    :param statename: State name which determines transition id
+    :param input: Input which contains a map from state to transition id
+    :param builder: Builder to append a call
+    :param parentstate: Optional name of parent state for parallel states.
```

Comment:
If parentState is only for parallel states, maybe this could be put into its name somehow?

Reply:
changed